src/components/GoogleMap/GoogleMap.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState, useEffect } from 'react';
+import GoogleMapReact, { fitBounds } from 'google-map-react';
+import { Checkbox } from 'antd';
+import { MapInterface, Bounds, Size } from './types';
+import { Marker } from './Marker';
+import styles from './GoogleMap.module.css';
+import robots from '../../MOCK_DATA.json';
+import markerImg from './Marker/pics/Marker.png';
+import { isPointInsideBounds, getCoordinatesForFitBounds } from './utils';
+
+const textOnCheckbox = 'Apply filter on bounds change';
+const DEFAULT_CENTER = { lat: 53.893009, lng: 27.567444 };
+const DEFAULT_ZOOM = 4;
+const DEFAULT_SIZE = { width: 1315, height: 800 };
+
+export const GoogleMap: React.FC<MapInterface> = ({
+  markers,
+  currentPosition,
+  setListOnBoundsChanged,
+  activeMarker,
+}) => {
+  const [isFilterOnBoundsChangeEnabled, setIsFilterOnBoundsChangeEnabled] = useState(true);
+  const [zoom, setZoom] = useState(DEFAULT_ZOOM);
+  const [center, setCenter] = useState(DEFAULT_CENTER);
+  const [mapSize, setMapSize] = useState(DEFAULT_SIZE);
+
+  const handleBoundsChange = ({ bounds, zoom, size }: { bounds: Bounds; zoom: number; size: Size }) => {
```

Comment:
For better readability better to  move `{ bounds: Bounds; zoom: number; size: Size }` to separate type